spec/requests/pages_spec.rb

```diff
@@ -0,0 +1,157 @@
+# frozen_string_literal: true
+
+describe "Pages", type: :request do
+  let(:user) { create(:user) }
+
+  describe "#home" do
+    it "respond to request" do
+      get pages_home_path
+      expect(response).to be_successful
+    end
+
+    context "logged in" do
+      before { sign_in user }
+
+      it "has no stories" do
+        list = double
+        expect(Kaminari).to receive(:paginate_array).and_return(list)
+        expect(list).to receive(:page)
+        get pages_home_path
+      end
+
+      it "has stories" do
+        create(:strategy, user_id: user.id)
+        categories = create_list(:category, 2, user_id: user.id)
+        moods = create_list(:mood, 2, user_id: user.id)
+        get pages_home_path
+        expect(assigns(:moment)).to be_a_new(Moment)
+        expect(assigns(:categories)).to eq(categories.reverse)
+        expect(assigns(:moods)).to eq(moods.reverse)
+      end
+    end
+
+    context "not logged in" do
+      it "has blurbs and posts" do
+        get pages_home_path
+        expect(assigns(:posts)[0].keys).to(
+          contain_exactly(:link, :link_name, :author)
+        )
+        blurbs_file = File.read("doc/pages/blurbs.json")
+        expect(assigns(:blurbs)).to eq(JSON.parse(blurbs_file))
+      end
+    end
+  end
+
+  describe "#home_data" do
+    let!(:moment) { create(:moment, user: user) }
+    context "when the user is logged in" do
+      before { sign_in user }
+      before { get home_data_path, params: {page: 1}, headers: {"ACCEPT" => "application/json"} }
+
+      it "returns a response with the correct path" do
+        expect(JSON.parse(response.body)["data"].first["link"]).to eq moment_path(moment)
+      end
+    end
+
+    context "when the user is not logged in" do
+      before { get home_data_path, params: {page: 1}, headers: {"ACCEPT" => "application/json"} }
+
+      it "returns a no_content status" do
+        expect(response).to have_http_status(:no_content)
+      end
+    end
+  end
+
+  describe "#partners" do
+    it "respond to request" do
+      get partners_path
+      expect(response).to be_successful
+    end
+
+    it "read external JSON file" do
+      data = []
+      file = File.read("doc/pages/partners.json")
+      expect(JSON).to receive(:parse).with(file).and_return(data)
+      expect(data).to receive(:sort_by!)
+      get partners_path
```

Comment:
The `expect().to receive()` are expectations for the future. It does the same as `allow().to receive()` but will fail if it is never called. This is not entirely intuitive, so there is also the `expect().to have_received()`

Reply:
Yeah, I was wondering since most patterns I see it would be like the following:

```ruby
# Arrange
allow().to receive()

# Act
get some_path

# Assert
expect().to have_received
```

Reply:
Ya, that's the pattern I've generally followed to, because most people see the other one and get confused. Not very intuitive, but it usually works the same way lol